circleci/resource_circleci_schedule.go

```diff
@@ -0,0 +1,268 @@
+		return err
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourceCircleCIScheduleRead(d *schema.ResourceData, m interface{}) error {
+	c := m.(*client.Client)
+	id := d.Id()
+
+	schedule, err := c.GetSchedule(id)
+	if err != nil {
+		return fmt.Errorf("Failed to read schedule: %s", id)
+	}
+
+	if schedule == nil {
+		d.SetId("")
+		return nil
+	}
+
+	_, organization, project, err := explodeProjectSlug(schedule.ProjectSlug)
+	if err != nil {
+		return err
+	}
+
+	d.Set("organization", organization)
+	d.Set("project", project)
+	d.Set("name", schedule.Name)
+	d.Set("description", schedule.Description)
+	d.Set("per_hour", schedule.Timetable.PerHour)
+	d.Set("hours_of_day", schedule.Timetable.HoursOfDay)
+	d.Set("days_of_week", schedule.Timetable.DaysOfWeek)
+	d.Set("parameters", schedule.Parameters)
+
+	if schedule.Actor.ID == scheduledActorID {
+		d.Set("use_scheduling_system", true)
+	} else {
+		d.Set("use_scheduling_system", false)
+	}
+
+	return nil
+}
+
+func resourceCircleCIScheduleUpdate(d *schema.ResourceData, m interface{}) error {
+	c := m.(*client.Client)
+
+	id := d.Id()
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	attributionActor := d.Get("use_scheduling_system").(bool)
+
+	parsedHours := d.Get("hours_of_day").([]interface{})
+	var hoursOfDay []uint
+	for _, hour := range parsedHours {
+		hoursOfDay = append(hoursOfDay, uint(hour.(int)))
+	}
+
+	var exists = struct{}{}
+	validDays := make(map[string]interface{})
+	validDays["MON"] = exists
+	validDays["TUE"] = exists
+	validDays["WED"] = exists
+	validDays["THU"] = exists
+	validDays["FRI"] = exists
+	validDays["SAT"] = exists
+	validDays["SUN"] = exists
+
+	parsedDays := d.Get("days_of_week").([]interface{})
+	var daysOfWeek []string
+	for _, day := range parsedDays {
+		if validDays[day.(string)] == nil {
+			return fmt.Errorf("Invalid day specified: %s", day)
+		}
+		daysOfWeek = append(daysOfWeek, day.(string))
+	}
+
+	timetable := api.Timetable{
+		PerHour:    uint(d.Get("per_hour").(int)),
+		HoursOfDay: hoursOfDay,
+		DaysOfWeek: daysOfWeek,
+	}
+
+	parsedParams := d.Get("parameters").(map[string]interface{})
+	parameters := make(map[string]string)
+	for k, v := range parsedParams {
+		parameters[k] = v.(string)
+	}
+
+	_, err := c.UpdateSchedule(id, name, description, timetable, attributionActor, parameters)
+	if err != nil {
+		return fmt.Errorf("Failed to update schedule: %w", err)
+	}
+
+	return nil
+}
+
+func resourceCircleCIScheduleImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	c := m.(*client.Client)
+
+	schedule, err := c.GetSchedule(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	d.SetId(schedule.ID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func explodeProjectSlug(slug string) (string, string, string, error) {
+	matches := strings.Split(slug, "/")
+
+	if len(matches) != 3 {
+		return "", "", "", fmt.Errorf("Splitting project-slug '%s' into vcs/org/project failed", slug)
```

Comment:
I feel like this error message counters itself. Maybe something along the lines of `Extracting vcs, org, and project from project-slug '%s' failed`? As it stands now it seems to imply that it's being split INTO `vcs/org/project` rather than the values being extracted